### DISCUSSION TITLE
Properly connecting deepgram for Speech to text/ text to speech to Audiocodes

### DISCUSSION THREAD

**Discussion by danielgold** [deepgram/community#1000] [General help] [answered]
Hi, 

I tried to connect Deepgram API to audiocodes for use of a phone AI agent. 

However, 

Even though it was set up inside Audiocodes, I was getting something called a 

TTS error 403. There doesn't seem to be any instructions anywhere on how to properly connect to Audiocodes, and there doesn't seem to be any obvious configurations inside deepgram.

**Comment by deepgram-community** [comment]
Hey there! It looks like you haven't connected your GitHub account to your Deepgram account. You can do this at https://community.deepgram.com - being verified through this process will allow our team to help you in a much more streamlined fashion.

**Comment by deepgram-community** [comment]
It looks like we're missing some important information to help debug your issue. Would you mind providing us with the following details in a reply?

- The programming language you are working in (e.g. JavaScript, Python).
- The deepgram product you are using (e.g Speech to Text, Agent API)
- A request ID that triggered your error or issue.

**Comment by danielgold** [comment]
Speech to text and API is what I'm using

**Comment by RyanChimienti** [comment, marked as answer]
Hey @danielgold , if you're using AudioCodes LiveHub, you can add Deepgram TTS and Deepgram STT as separate services under `Speech Services`. Both can be `Generic provider (using AC API)`. The URLs are:

- TTS: `https://integrations.deepgram.com/audiocodes/tts`
- STT: `wss://integrations.deepgram.com/audiocodes/stt`

This TTS configuration works for me:
![image](https://github.com/user-attachments/assets/ed760666-6cd0-4ec6-a028-fc6ee2abfbe7)

Since you're getting a 403 error, another thing you should do is validate that your API key can access Deepgram's TTS API. You can try a request like this (swapping in your API key) and make sure you get a playable mp3 file back:
```
curl -X POST \
--url 'https://api.deepgram.com/v1/speak?model=aura-asteria-en' \
-H "Authorization: Token $DEEPGRAM_API_KEY" \
-H "content-type: application/json" \
-d '{"text": "Hey there!"}' > test.mp3
```
